nats/src/main.rs

```diff
@@ -485,6 +473,64 @@ fn should_strip_headers(topic: &str) -> bool {
     topic.starts_with("$SYS")
 }
 
+fn build_connect_options(cfg: &ConnectionConfig) -> Result<async_nats::ConnectOptions, RpcError> {
+    let opts = match (cfg.auth_jwt.clone(), cfg.auth_seed.clone()) {
+        (Some(jwt), Some(seed)) => {
+            let key_pair = std::sync::Arc::new(
+                KeyPair::from_seed(&seed)
+                    .map_err(|e| RpcError::ProviderInit(format!("key init: {}", e)))?,
+            );
+            async_nats::ConnectOptions::with_jwt(jwt, move |nonce| {
+                let key_pair = key_pair.clone();
+                async move { key_pair.sign(&nonce).map_err(async_nats::AuthError::new) }
+            })
+        }
+        (None, None) => async_nats::ConnectOptions::default(),
+        _ => {
+            return Err(RpcError::InvalidParameter(
+                "must provide both jwt and seed for jwt authentication".into(),
+            ));
+        }
+    };
+
+    if let Some(tls_ca) = &cfg.tls_ca {
+        return add_tls_ca(tls_ca, opts);
+    }
+
+    Ok(opts)
+}
+
+fn add_tls_ca(
+    tls_ca: &str,
+    opts: async_nats::ConnectOptions,
+) -> Result<async_nats::ConnectOptions, RpcError> {
+    let mut store_builder = async_nats::rustls::RootCertStore::empty();
+    let parsed_ca = rustls_pemfile::read_one(&mut tls_ca.as_bytes())
+        .map_err(|e| RpcError::ProviderInit(format!("tls ca: {}", e)))?;
+    if let Some(parsed) = parsed_ca {
+        match parsed {
+            rustls_pemfile::Item::X509Certificate(cert) => {
+                store_builder.add_parsable_certificates(&[cert]);
+            }
+            _ => {
+                return Err(RpcError::ProviderInit(
+                    "tls ca: invalid certificate type, must be a DER encoded PEM file".into(),
+                ));
+            }
+        }
+
+        let tls_client = async_nats::rustls::ClientConfig::builder()
```

Comment:
Nit: since the other branches all return, moving this to the end of the function would make it easier to read